src/platforms/ctxlink/winc1500_driver_stub_api.c

```diff
@@ -0,0 +1,265 @@
+
+// A TMR peripheral can be set to interrupt every 1ms. The timer ISR increments
+// a global counter.
+
+static _Atomic uint32_t one_ms_counter = 0;
+
+//
+// Implement this timer ISR or call this function in your timer ISR to
+// increment one_ms_counter variable as below:
+//
+/* NOLINTNEXTLINE(readability-identifier-naming) */
+void m2m_TMR_ISR(void)
+{
+	one_ms_counter++;
+}
+
+/* NOLINTNEXTLINE(readability-identifier-naming) */
+uint32_t m2mStub_GetOneMsTimer(void)
+{
+	return one_ms_counter;
+}
+
+//==============================================================================
+// SPI Stub Functions:
+// ---------------------
+//    - The Host MCU will communicate to the WINC1500 via the SPI interface.
+//==============================================================================
+
+/* NOLINTNEXTLINE(readability-identifier-naming) */
+void m2mStub_SpiTxRx(uint8_t *p_txBuf, uint16_t txLen, uint8_t *p_rxBuf, uint16_t rxLen)
+{
+	uint16_t byteCount;
+	//
+	// The following are intermediate buffers used to ensure
+	// the TX and RX message sizes handled by the transfer routine
+	// are the same size.
+	//
+	uint8_t outputBuffer[SPI_TRANSFER_BUFFER_SIZE] = {0};
+	uint8_t inputBuffer[SPI_TRANSFER_BUFFER_SIZE] = {0};
+	/*
+     *	total number of byte to clock is whichever is larger, txLen or rxLen
+     */
+	byteCount = txLen >= rxLen ? txLen : rxLen;
+	if (txLen > SPI_TRANSFER_BUFFER_SIZE || rxLen > SPI_TRANSFER_BUFFER_SIZE)
+		return; // No way to communicate error, silently ignore data
+	//
+	// Copy the input data to the outputBuffer
+	//
+	memcpy(&outputBuffer[0], p_txBuf, txLen);
+	//
+	// Do the transfer
+	//
+	for (uint16_t i = 0; i < byteCount; i++)
+		inputBuffer[i] = spi_xfer(WINC1500_SPI_CHANNEL, outputBuffer[i]);
+	//
+	// If we expected to receive bytes copy them to the rx buffer
+	//
+	if (rxLen > 0)
+		memcpy(p_rxBuf, &inputBuffer[0], rxLen);
+}
+
+//==============================================================================
+// Event Stub Functions:
+// ---------------------
+//    These are callback functions that the WINC1500 host driver calls to notify
+//    the application of events. There are four categories of events:
+//      - Wi-Fi events
+//      - Socket events
+//      - OTA (Over-The-Air) update events
+//      - Error Events
+//==============================================================================
+
+tpfAppWifiCb app_wifi_cb = NULL;
+
+/* NOLINTNEXTLINE(readability-identifier-naming) */
+void registerWifiCallback(tpfAppWifiCb pfAppWifiCb)
+{
+	app_wifi_cb = pfAppWifiCb;
+}
+
+void m2m_wifi_handle_events(t_m2mWifiEventType eventCode, t_wifiEventData *p_eventData)
+{
+	if (app_wifi_cb)
+		app_wifi_cb(eventCode, p_eventData);
+	else
+		DEBUG_WARN("STUB_WIFI_EVENT[%d]: Wi-Fi event handler not registered!\r\n", eventCode);
+}
+
+//             --------------- * end of wifi event block * ---------------
+
+volatile tpfAppSocketCb app_socket_cb = NULL;
```

Comment:
Same comment as with the `app_wifi_cb` originally being volatile.

All global variable definitions like these should go at the top of the file please.

Reply:
Changed as requested.